src/Mapping/Table.php

```diff
@@ -21,5 +22,24 @@ public function __construct(
         public readonly array|null $uniqueConstraints = null,
         public readonly array $options = [],
     ) {
+        if ($this->indexes !== null && $this->indexes !== []) {
```

Comment:
True its unnecessary since the deprecation should be triggered as soon as something is passed at all.

I've changed this and added the requested test for it.